code/modules/food_and_drink/pies.dm

```diff
@@ -9,12 +9,42 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food/snacks/pie)
 	var/slicetype = /obj/item/reagent_containers/food/snacks/pieslice
 	var/splat = 0 // for thrown pies
 	food_effects = list("food_refreshed","food_cold")
+	///In the case of a thrown splattered pie, minimum amount of time we remain visually stuck on someone's face.
+	var/min_stuck_time = 5 SECONDS
+	///In the case of a thrown splattered pie, maximum amount of time we remain visually stuck on someone's face.
+	var/max_stuck_time = 10 SECONDS
 
 	throw_impact(atom/hit_atom, datum/thrown_thing/thr)
 		if (ismob(hit_atom) && src.splat)
 			var/mob/M = hit_atom
-			src.visible_message("<span class='alert'>[src] splats in [M]'s face!</span>")
+			var/mob/thrower = thr.thrown_by
 			playsound(src, 'sound/impact_sounds/Slimy_Splat_1.ogg', 100, 1)
+			if (thrower.mind?.assigned_role == "Clown")
+				if (ishuman(M) && (prob(50) || M.mind?.assigned_role == "Captain") && !M.GetOverlayImage("face_pie"))
+					var/mob/living/carbon/human/H = M
+					var/image/face_pie = image('icons/obj/foodNdrink/food_dessert.dmi', "face_pie")
+					src.visible_message("<span class='notice'>[src] splats right in [H]'s face and remains stuck there!</span>")
+					face_pie.layer = MOB_OVERLAY_BASE
+					face_pie.appearance_flags = RESET_COLOR | PIXEL_SCALE
+					var/overlay_key = "face_pie"
+					if(H.mutantrace.head_offset)
+						face_pie.pixel_y = H.mutantrace.head_offset
+					M.UpdateOverlays(face_pie, overlay_key)
+					src.set_loc(M)
+					M.bioHolder?.AddEffect("bad_eyesight")
+					JOB_XP(thrower, "Clown", 1)
+					SPAWN(rand(src.min_stuck_time, src.max_stuck_time))
+						if (QDELETED(M))
+							return
+						M.bioHolder?.RemoveEffect("bad_eyesight")
+						M.UpdateOverlays(null, overlay_key)
+						if (QDELETED(src))
+							return
+						src.visible_message("<span class='notice'>[src] falls off of [M]'s face.</span>")
+						src.set_loc(M.loc)
+						qdel(face_pie)
+					return
+			src.visible_message("<span class='alert'>[src] splats in [M]'s face!</span>")
```

Comment:
This should probably be in an `else` so you don't get two messages when a clown throws a pie

Reply:
Donezo